environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/azure/AzureEnvironmentSecurityGroupValidator.java

```diff
@@ -59,18 +66,28 @@ private void validateSecurityGroup(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupFoundInRegion = false;
-        if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())
-                && Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses().get(region.getName()))) {
-            for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-                String groupId = cloudSecurityGroup.getGroupId();
-                if (groupId.equalsIgnoreCase(securityGroupId)) {
-                    securityGroupFoundInRegion = true;
+        Map<String, Set<CloudSecurityGroup>> cloudSecurityGroupsResponses = securityGroups.getCloudSecurityGroupsResponses();
+        if (Objects.nonNull(cloudSecurityGroupsResponses)) {
+            Set<CloudSecurityGroup> cloudSecurityGroups = securityGroups.getCloudSecurityGroupsResponses().get(region.getName());
```

Comment:
Nit: `securityGroups.getCloudSecurityGroupsResponses()` could be replaced with `cloudSecurityGroupsResponses`.